.github/workflows/build.yml

```
@@ -2,13 +2,17 @@ on:
   push:
     branches:
       - main
+  pull_request:
 jobs:
   mac_build:
     runs-on: macos-latest
     steps:
     - uses: actions/checkout@v1
     - run: brew install python@3.9    # https://github.com/actions/setup-python/issues/58
-    - run: pip3 install pyinstaller appdirs
+    # TODO: is it possible to use a newer pyinstaller?
+    # --no-use-pep517 is to prevent pyproject.toml error when installing old pyinstaller.
+    - run: pip3 install --no-use-pep517 pyinstaller==4.10
+    - run: pip3 install appdirs
```

Lets try without the --no-use-pep517 flag

```suggestion
    - run: pip3 install pyinstaller==4.10 appdirs
```